lib/matplotlib/backends/backend_qt4.py

```diff
@@ -1,48 +1,16 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
-import six
-from six import unichr
-import os
-import re
-import signal
-import sys
-
-import matplotlib
-
-from matplotlib.cbook import is_string_like
-from matplotlib.backend_bases import FigureManagerBase
 from matplotlib.backend_bases import FigureCanvasBase
-from matplotlib.backend_bases import NavigationToolbar2
-
-from matplotlib.backend_bases import cursors
-from matplotlib.backend_bases import TimerBase
-from matplotlib.backend_bases import ShowBase
-
-from matplotlib._pylab_helpers import Gcf
 from matplotlib.figure import Figure
 
+from .qt_compat import QtCore, QtWidgets, __version__
 
-from matplotlib.widgets import SubplotTool
-try:
-    import matplotlib.backends.qt_editor.figureoptions as figureoptions
-except ImportError:
-    figureoptions = None
-
-from .qt_compat import QtCore, QtWidgets, _getSaveFileName, __version__
-from matplotlib.backends.qt_editor.formsubplottool import UiSubplotTool
-
-from .backend_qt5 import (backend_version, SPECIAL_KEYS, SUPER, ALT, CTRL,
```

Comment:
If there is an __all__ in the module, then I would say that it is safe to
throw out anything not listed in the __all__ that isn't needed anymore.
But, yeah, this is what happens when you become a major package in a
language and you aren't careful when you create modules. You learn the
lesson and try and do better moving forward.

On Mon, Apr 20, 2015 at 5:05 PM, Kelvin Li <notifications@github.com> wrote:

> In lib/matplotlib/backends/backend_qt4.py
> <https://github.com/matplotlib/matplotlib/pull/4363#discussion_r28729127>:
>
> >
> > -from matplotlib.widgets import SubplotTool
> > -try:
> > -    import matplotlib.backends.qt_editor.figureoptions as figureoptions
> > -except ImportError:
> > -    figureoptions = None
> > -
> > -from .qt_compat import QtCore, QtWidgets, _getSaveFileName, __version__
> > -from matplotlib.backends.qt_editor.formsubplottool import UiSubplotTool
> > -
> > -from .backend_qt5 import (backend_version, SPECIAL_KEYS, SUPER, ALT, CTRL,
>
> Thanks for the tips. Just for my own understanding, does this mean we
> cannot break users who do the following?
>
> from matplotlib.backends.backend_qt4 import SubplotTool
>
> I'm surprised that this is considered part of the external API, but I'll
> take your word for it.
>
> Assuming we are on the same page, then moving forward, I'm not even sure
> what is considered safe to prune... any advice?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4363/files#r28729127>.
>